File: examples/eApi v1.9/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/Act/AuthData.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects.Act;

public class AuthData : ApiBase
{
    [JsonProperty("browser")] public Auth3dsBrowser? Browser { get; set; }

    [JsonProperty("sdk")] public Auth3dsSdk? Sdk { get; set; }


    public string? ToSign()
    {
        var sb = new StringBuilder();
        if (null != Browser) sb.Append(Browser.ToSign());
        if (null != Sdk) sb.Append(Sdk.ToSign());
        DeleteLast(sb);
        return sb.ToString();
    }
}

File: examples/eApi v1.9/java/eAPI-v19-connector/src/main/java/cz/monetplus/mips/eapi/v19/connector/entity/AuthData.java
package cz.monetplus.mips.eapi.v19.connector.entity;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
@JsonInclude(value = JsonInclude.Include.NON_NULL)
public class AuthData extends ApiBase implements Signable {
        private Auth3dsBrowser browser;
        private Auth3dsSdk sdk;

        @Override
        public String toSign() {
            StringBuilder sb = new StringBuilder();
            if (null != browser) sb.append(browser.toSign());
            if (null != sdk) sb.append(sdk.toSign());
            deleteLast(sb);
            return sb.toString();
        }
    }